nomt/tests/extend_range_protocol.rs

```diff
@@ -0,0 +1,68 @@
+mod common;
+use common::Test;
+use std::path::Path;
+
+#[rustfmt::skip]
+const KEYS_AND_VALUE_SIZES: [(u8, usize); 16] =[
```

Comment:
this seems like a good approach. but it is kind of fragile to the page size (though a lot of our code is). looks good, but add a comment explaining why this forces new leaves!

Reply:
Yes! Those numbers are tightly related to the size of a node/page! I will add a comment and the assertion!

Reply:
well, unfortunately both `BRANCH_NODE_SIZE` and `PAGE_SIZE` are inside private modules, not reachable from integration testing. I will add a clear comment explaining the dependency from the size of the node

Reply:
perfect